src/applications/caregivers/components/PreSubmitInfo/SubmitLoadingIndicator.jsx

```diff
@@ -0,0 +1,72 @@
+import React, { useEffect, useState, useCallback } from 'react';
+import { connect } from 'react-redux';
+import LoadingIndicator from '@department-of-veterans-affairs/component-library/LoadingIndicator';
+
+const SubmitLoadingIndicator = ({ submission }) => {
+  const [isLoading, setLoading] = useState(false);
+  const [timer, setTimer] = useState(0);
+  const [loadingMessage, setLoadingMessage] = useState(null);
+  const hasAttemptedSubmit = submission.hasAttemptedSubmit;
+  const isSubmitPending = submission.status === 'submitPending';
+
+  const getLoadingMessage = useCallback(
+    () => {
+      if (timer >= 0) {
+        setLoadingMessage('Please wait while we process your application.');
+      }
+
+      if (timer >= 15) {
+        setLoadingMessage(
+          'We’re processing your application. This may take up to 1 minute. Please don’t refresh your browser.',
+        );
+      }
+    },
+    [timer],
+  );
+
+  useEffect(
+    () => {
+      if (hasAttemptedSubmit && isSubmitPending) {
+        setLoading(true);
+      }
+    },
+    [hasAttemptedSubmit, isSubmitPending],
+  );
+
+  useEffect(
+    () => {
+      let interval = null;
+
+      if (isLoading) {
+        interval = setInterval(() => {
+          setTimer(seconds => seconds + 1);
```

Comment:
Is this being used anywhere else? It doesn't look like it's added to the loading message any more, can this be removed?

Reply:
Is what being used the timer? It is defiantly still being used as dependencies in the useEffect and useCallback that makes the content changes on a specific time during the spinner. In this case the content changes at 15 seconds.

Reply:
Oh, I meant `seconds` is only used here and no where else in this file.

Reply:
Its a callback in set timer that is like the prevState

Reply:
Oh duh... it sets `timer`... sorry, lots of distractions this morning!